src/System.Numerics.Vectors/tests/Performance/Vector2/DistanceSquared.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+
+namespace System.Numerics.Tests
+{
+    public static partial class Perf_Vector2
+    {
+        [Benchmark(InnerIterationCount = VectorTests.DefaultInnerIterationsCount)]
+        public static void DistanceSquaredBenchmark()
+        {
+            const float expectedResult = 134217728.0f;
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                float result;
+
+                using (iteration.StartMeasurement())
+                {
+                    result = DistanceSquaredTest();
+                }
+
+                VectorTests.AssertEqual(expectedResult, result);
+            }
+        }
+
+        public static float DistanceSquaredTest()
+        {
+            var result = 0.0f;
+            var valueX = new Vector2(-1.0f, 1.0f);
+            var valueY = new Vector2(1.0f, -1.0f);
+
+            for (var iteration = 0; iteration < Benchmark.InnerIterationCount; iteration++)
+            {
+                valueX += VectorTests.Vector2Delta;
```

Comment:
So I did a few experiments and found a few interesting things.  So for a test like `DistanceSquared` removing the two adds from the timing loop first and foremost does not result in the entire method being elided by the JIT.  Second, the difference between no adds and the two adds that we currently have is a 50% difference.  We go from 2591ms to 3889ms on my box for one billion iterations.  For `Vector3` we are only seeing a 13% difference, but even that is big enough that we movement in the add will have multiple percent impacts on timing for other scenarios.

Given that we do not need the additional adds to ensure that the JIT does not remove our code entirely I believe that we should remove all of the Vector adds inside the timing loops.